### DISCUSSION TITLE
An effect returned for this action is still running.

### DISCUSSION THREAD

**Discussion by DispoKen** [pointfreeco/swift-composable-architecture#967] [Q&A]
Hi, I have watched the video regarding this specific error message, however I still have no idea how to fix this error. In our test, we seemed to have covered all the possible effects, but we still end up getting this message. Is there a way to find what remaining effects that are still needed to address? I notice the error message sometimes does give us a helpful effect to look at, but for our error message it doesn't give any real hints aside from checking for long living effects which is hard to find. Any advice in helping to debug this is greatly appreciated! 
<img width="2013" alt="Screen Shot 2022-01-13 at 7 18 47 PM" src="https://user-images.githubusercontent.com/89225304/149432750-825e3ca4-c91a-40e7-8f86-49adffeefb9d.png">

**Comment by leovido** [comment]
Looks like you might need to add `.receive(environment.mainQueue)` for the effect that runs in your `.onAppear` action.

`case .onAppear:
    return Effect(value: SomeAction.foo)
                    .receive(environment.mainQueue)`

**Comment by stephencelis** [comment]
This error means that one of the effects returned from `.onAppear` hasn't completed. So if you've covered effects you expect to fire, there still may be a long-living effect that needs to be cancelled by sending another action, like `.onDisappear`. Can you share what the effects the reducer(s) that handle `.onAppear` look like for that case? That might help us troubleshoot.
